stix_shifter_modules/splunk/stix_transmission/results_connector.py

```diff
@@ -1,29 +1,214 @@
-        response = await self.api_client.get_search_results(search_id, offset, length)
-        response_code = response.code
-        response_dict = json.load(response)
-
-        # Construct a response object
-        return_obj = dict()
-        if response_code == 200:
-            if "results" in response_dict:
-                results = response_dict['results']
+        return_obj = {}
+        response_dict = {}
+        try:
+            response = await self.api_client.get_search_results(search_id, offset, length)
+            response_code = response.code
+            response_dict = json.load(response)
+            response_text = response.content
+            
+            # Construct a response object
+            return_obj = {}
+            if response_code == 200:
+                if "results" in response_dict:
+                    results = response_dict['results']
+                else:
+                    results = []
+                if results != []:
+                    results = ResultsConnector.check_data(results)
+                else:
+                    results = []
+                return_obj['success'] = True
+                return_obj['data'] = results
             else:
-                results = []
-            return_obj['success'] = True
-            return_obj['data'] = results
-        else:
-            ErrorResponder.fill_error(return_obj, response_dict, ['messages', 0, 'text'], connector=self.connector)
+                response_dict['type'] = str(response_code)
+                if response_code == 404:
+                    response_dict['type'] = "Unknown_sid"
+                response_dict['message'] = response_text
+                ErrorResponder.fill_error(return_obj, response_dict, ['messages'],
+                                          connector=self.connector)
+        except ClientConnectionError:
+            response_dict['type'] = "ConnectionError"
+            response_dict['message'] = "Invalid Host"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except TimeoutError as ex:
+            response_dict['type'] = "Timeout"
+            response_dict['messages'] = "TimeoutError"
+            ErrorResponder.fill_error(return_obj, response_dict, ['messages'], connector=self.connector)
+        except Exception as ex:
+            if 'Authentication error' in str(ex):
+                response_dict['type'] = "AuthenticationError"
+            elif 'timeout_error' in str(ex):
+                response_dict['type'] = "Timeout"
+            response_dict['messages'] = str(ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['messages'], connector=self.connector)
+
         return return_obj
+
+    @staticmethod
+    def check_data(res_data):
+        """
+        check data and update the api response as required
+        :param item: list of dictionary items
+        :return: list
+        """
```

Comment:
can you try using transformers for some of the value checking? for example: registry value, severity

Reply:
> severity

In results_connector, we are only preprocessing the responses and not converting severity and registry values.
We are using SeverityToScore and RegistryValueName transformers for severity and registry conversion.